### PR TITLE
Feature/pseudocode unit tests

### PR DESCRIPTION
**Pull request by chadrick-d-dev**
## Description

- We wrote pseudocode for unit and integration testing all of our app components (App, MovieContainer, SingleMovie, MovieCard, Header)

## Motivation and Context

- Pseudocode was required as part of iteration 2.

## How Has This Been Tested?

- No testing, pseudocode only 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Tickets/Issues
- #14 #15 #16 #17 #18

### CODE REVIEW

File: src/App/App.test.js [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+import '@testing-library/jest-dom';
 import App from './App';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+describe('App', () => {
+  it('should take user to movie cards view when return to home button is clicked', () => {
+    // setup
```
Comment (saraho1123) [inline src/App/App.test.js:7]: We need to do a little research about whether to do tests for methods on the component where they are defined or where they are invoked. Either way, I think our pseudocode is a good start! :)

File: src/Header/Header.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import Header from './Header';
+
+describe('Header', () => {
+  it('should return a header when page loads', () => {
+    // setup
+    // we want to create a render that renders Header
+    // execution
+    // not needed ?
+    // assertion
+    // we want to look for 'Rancid Tomatillos Movie ratiangs' in the document
```
Comment (saraho1123) [inline src/Header/Header.test.js:13]: I think it is implied, but just want to make sure we remember the queries!

File: src/App/App.test.js [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
 import App from './App';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+describe('App', () => {
+  it('should take user to movie cards view when return to home button is clicked', () => {
```
Comment (chadrick-d-dev) [inline src/App/App.test.js:6]: Definitely want to see if we are doing this test in right test file. When we run our tests we'll find out!

File: src/MovieCard/MovieCard.test.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  it('should return a movie card', () => {
+    // setup
+    // render of SingleCard, with mocked, id, image, title, rating, and displayMovieDetails
+    // execution
+    // 
+    // assertion
+    // we need assert that title and rating and an image are in the document
+  })
+
+  it('should take user to single movie when card is clicked', () => {
```
Comment (saraho1123) [inline src/MovieCard/MovieCard.test.js:15]: Ah ... we left this pseudocode here and in App. That works until we do a bit more research!

File: src/SingleMovie/SingleMovie.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    // we need to render SingleMovie, making sure to include the movieDetails and returnHome in the render
+    // execution
+    // 
+    // assertion
+    // assert that all the movie details and existing inner text display with 
+    // use of expect toBeInTheDocuments
+  })
+
+  // Integration test for app
+  it('should take user to movie cards view when return to home button is clicked', () => {
```
Comment (saraho1123) [inline src/SingleMovie/SingleMovie.test.js:20]: And we have this one both here and in App, so we are good to go after we do the research!

**Review by saraho1123 — APPROVED** [review]
Awesome! It is great to take time to pseudocode!! I think we will need to get some more sad path testing, but at least we have a start! :)

**Pull request merged**